src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -644,4 +649,44 @@ private List<Long> parseSupplementalGroupList(String gids) {
         }
         return Collections.unmodifiableList(builder);
     }
+
+    /**
+     * <p>
+     * {@link PodDecorator} allowing to inject in {@code jnlp} containers definition a {@code securityContext} definition allowing to use the
+     * {@code restricted} <a href="https://kubernetes.io/docs/concepts/security/pod-security-standards/">Pod Security Standard</a>.
+     * </p>
+     * <p>
+     * See <a href="https://issues.jenkins.io/browse/JENKINS-71639">JENKINS-71639</a> for more details.
+     * </p>
+     */
+    @Extension
+    public static class RestrictedPssSecurityContextInjector implements PodDecorator {
+
+        @NonNull
+        @Override
+        public Pod decorate(@NonNull KubernetesCloud kubernetesCloud, @NonNull Pod pod) {
+            if (kubernetesCloud.isRestrictedPssSecurityContext()) {
+                Optional<Container> maybeJNLP = pod.getSpec().getContainers().stream().filter(container -> JNLP_NAME.equals(container.getName())).findFirst();
+
+                maybeJNLP.ifPresent(jnlp -> {
```

Comment:
If the `jnlp` container is not defined at this point then there is a bug somewhere. At this point this code should assume so, and if the container is not defined yet then print a SEVERE error log (as written it's just silently ignoring the problem).

Reply:
81226b76